dill/settings.py

```diff
@@ -19,7 +21,10 @@
     'fmode' : 0, #HANDLE_FMODE
     'recurse' : False,
     'ignore' : False,
+    'dump_module' : {
```

Comment:
I don't really like the idea of a nested settings dict.  To me, the idea of `settings` is that it can be used to override the similarly named `kwargs` on certain pickling functions.  If `dump_module` is used as a key, why then not `dump` and `load`?  As long as the keys are unique, it shouldn't matter if it's a flat dict (the old `byref` <- `refimported` broke this).

Reply:
I understand these parameter defaults in `settings` are general in spirit, i.e. they apply to all the main functions and methods where it makes sense to apply them (and exceptions, like `recurse` for `dump_module` are now documented). With these new settings there would be more exceptions than cases...

What about renaming the `dump_module` entry, to `session`? It matches the module name and would still make sense —except that someone may think that filters would somehow apply to `load_module()`. If a nested dictionary is the main problem, I suggest using `session.refimported`, `session.refonfail` and `session.filters` as keys directly in the `settings` dictionary.

Reply:
I like the "per module" idea. What about `dill.session.settings`, but not included in the `dill.settings` dict?

Reply:
@mmckerns Does my previous comment was appearing for you? It was in a "pending review" state. I don't know if that means it's like a draft and was visible just for me or if it was public.

Reply:
Moved to `dill.session.settings`. Pending changes in #475